### PR TITLE
YUG-4241: Added new automation GRAAnomaliesDisplay.py, Corrected desc…

### PR DESCRIPTION
**Pull request by Sudhi108**
…riptions of all commands, fixed some typo errors in description.

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [x] 6.1.0
- [x] 6.2.0
- [x] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by adi88d** [comment]
@Sudhi108 please fill the [Contribution Registration](https://forms.gle/XDfxU4E61ZwEESSMA) form

**Comment by Sudhi108** [comment]
@adi88d I had filled earlier also when submitted the pull request and filled it again now. 
Thanks.

**Review by adi88d — CHANGES_REQUESTED** [review]
Hi @Sudhi108,
Thank you for your contribution!
Good work :)
see my comments below

File: Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.py [hunk 1]
```diff
@@ -334,6 +334,11 @@ def main() -> None:
             if entityTypeId is not None:
                 analyticalFeatures_url += '&entityTypeId=' + entityTypeId
             fetch_records(client, analyticalFeatures_url, 'Gra.Analytical.Features.Entity.Value', 'entityID', params)
+        elif demisto.command() == 'gra-cases-anomaly':
+            caseId = arguments.get('caseId')
+            anomaliesUrl = '/anomalies/' + caseId
+            # print(base_url)
```
[the remark this answers is not in the document]
Comment (Sudhi108) [reply Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.py:340]: removed

File: Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.yml [hunk 2]
```diff
@@ -949,8 +949,17 @@ script:
...
-  dockerimage: demisto/python3:3.10.8.37233
+    description: Retrieve analytical features for specified entity value, model name and dates.
+  - name: gra-cases-anomaly
+    arguments:
+    - name: caseId
+      description: GRA Case Id
+    outputs:
+    - contextPath: Gra.Cases.anomalies
+      description: Cases Anomalies
+      type: Unknown
```
[the remark this answers is not in the document]
Comment (Sudhi108) [reply Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.yml:960]: added context path to all outputs

File: Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.yml [hunk 3]
```diff
@@ -949,8 +949,17 @@ script:
     - contextPath: Gra.Analytical.Features.Entity.Value.analyticalFeatureValues
       description: Analytical Feature Values
       type: string
-    description: Retrieve analytical features for specified entity value and model name.
-  dockerimage: demisto/python3:3.10.8.37233
+    description: Retrieve analytical features for specified entity value, model name and dates.
+  - name: gra-cases-anomaly
+    arguments:
+    - name: caseId
+      description: GRA Case Id
```
[the remark this answers is not in the document]
Comment (Sudhi108) [reply Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.yml:956]: added required tag

File: Packs/Gurucul/Integrations/GuruculGRA/README.md [hunk 4]
```diff
@@ -1078,6 +1078,55 @@ Retrieve analytical features for specified entity value and model name.
...
+
+| **Argument Name** | **Description** | **Required** |
+|-------------------|-----------------| --- |
+| caseId            | GRA Case Id     | Required | 
+
+#### Context Output
+
+| **Path** | **Type** | **Description**    |
+| --- |----------|--------------------|
+|Gra.Cases.anomalies | Grid     | GRA Case Anomalies |
```
[the remark this answers is not in the document]
Comment (Sudhi108) [reply Packs/Gurucul/Integrations/GuruculGRA/README.md:1100]: done

File: Packs/Gurucul/ReleaseNotes/2_0_0.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+- Changed the Gurucul-GRA logo
+- Changed the description of all commands
+
+
+#### Layouts
+##### GRACaseLayout
+- Added new dynamic section to incident layout
+- This section uses the new automation script GRAAnomaliesDisplay
+- With this change, XSOAR will be able to get the newly added/modified anomalies for a particular case/incident from GRA
+- The new anomalies will then be synced with XSOAR
```
[the remark this answers is not in the document]
Comment (Sudhi108) [reply Packs/Gurucul/ReleaseNotes/2_0_0.md:24]: If I remove this, it shows as one message

File: Packs/Gurucul/Scripts/GRAAnomaliesDisplay/GRAAnomaliesDisplay.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
+import demistomock as demisto
+from CommonServerPython import *
+
+
+def _get_incident():
+    return demisto.incident()
+
+
+def getAnomaliesByCaseId():
+    incident = _get_incident()
```
[the remark this answers is not in the document]
Comment (Sudhi108) [reply Packs/Gurucul/Scripts/GRAAnomaliesDisplay/GRAAnomaliesDisplay.py:10]: fixed

File: Packs/Gurucul/Scripts/GRAAnomaliesDisplay/README.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
+Retrieve anomalies for specified case id from GRA and update in XSOAR
+
+## Script Data
+---
+
+| **Name** | **Description** |
+| --- | --- |
+| Script Type | python3 |
+| Tags | dynamic-section |
+| Cortex XSOAR Version | 6.5.0 |
```
[the remark this answers is not in the document]
Comment (Sudhi108) [reply Packs/Gurucul/Scripts/GRAAnomaliesDisplay/README.md:10]: fixed

**Comment by Sudhi108** [comment]
Hi @adi88d,

Thank you for such a detailed review.
I have made the changes as per your comments and pushed in git.

File: Packs/Gurucul/ReleaseNotes/2_0_0.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (adi88d) [reply Packs/Gurucul/ReleaseNotes/2_0_0.md]: @Sudhi108 it looks the same:
image

**Comment by adi88d** [comment]
Nice

> Hi @adi88d,
> 
> Thank you for such a detailed review. I have made the changes as per your comments and pushed in git.

just update the file `Packs/Gurucul/ReleaseNotes/2_0_0.md` and we good to go

**Review by adi88d — APPROVED** [review]

**Pull request merged**